features/dd-sdk-android-trace/src/main/kotlin/com/datadog/android/trace/internal/RumContextPropagator.kt

```diff
@@ -0,0 +1,141 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+package com.datadog.android.trace.internal
+
+import com.datadog.android.api.InternalLogger
+import com.datadog.android.api.context.DatadogContext
+import com.datadog.android.api.feature.Feature
+import com.datadog.android.api.feature.FeatureSdkCore
+import com.datadog.android.api.feature.getContextFuture
+import com.datadog.android.lint.InternalApi
+import com.datadog.android.log.LogAttributes
+import com.datadog.android.trace.api.span.DatadogSpan
+import com.datadog.android.trace.api.span.DatadogSpanBuilder
+import com.datadog.trace.core.DDSpan
+import java.util.concurrent.Future
+import java.util.concurrent.TimeUnit
+
+/**
+ * For internal usage only.
+ *
+ * This is a utility class that manages the propagation of RUM contexts for span events and HTTP tracing.
+ */
+@InternalApi
+class RumContextPropagator(private val sdkCoreProvider: () -> FeatureSdkCore?) {
+
+    private val sdkCore: FeatureSdkCore?
+        get() = sdkCoreProvider()
+
+    private val internalLogger: InternalLogger?
+        get() = sdkCore?.internalLogger
+
+    private fun injectRumContext(builder: DatadogSpanBuilder) {
+        sdkCore?.getFeature(Feature.RUM_FEATURE_NAME)
+            ?.getContextFuture(withFeatureContexts = setOf(Feature.RUM_FEATURE_NAME))
```

Comment:
question: is there advantage for creating `getContextFuture` method instead of using `withContext`? Seems like we added a method to the core API just for a single use-case, where `withContext` can be used.

Reply:
Yes, but then you need to do type-check for `SdkFeature`, because it is not a part of interface (which is for good).

The goal is to have a limited ways to work with context, to simplify the DevEx, like a single simple thing which do the stuff, otherwise we may create a problem of choice.

But I'm ok with having it like that, given it doesn't make it to the interface.